doc/kb_sdk_make_ui.md

```diff
@@ -0,0 +1,148 @@
+warned if a name with overwrite an existing object or if the name contains invalid characters.
+
+```
+	"parameters": [ 
+		{
+			"id": "read_library_name",
+			"optional": false,
+			"advanced": false,
+			"allow_multiple": false,
+			"default_values": [ "" ],
+			"field_type": "text",
+			"text_options": {
+				"valid_ws_types": ["KBaseAssembly.PairedEndLibrary","KBaseFile.PairedEndLibrary"]
+			}
+		},
+		{
+		    "id" : "output_contigset_name",
+		    "optional" : false,
+		    "advanced" : false,
+		    "allow_multiple" : false,
+		    "default_values" : [ "MegaHit.contigs" ],
+		    "field_type" : "text",
+		    "text_options" : {
+		     	"valid_ws_types" : [ "KBaseGenomes.ContigSet" ],
+		    	"is_output_name":true
+		    }
+		}
+	],
+```
+Another common input is a dropdown which is demonstrated below. For each option, the "value" is what will be passed to
+the script while the UI Name and Display is what the user will see. In this example, the parameter is hidden be default 
+because "advanced" is true.
+```
+{
+            "id": "prune",
+            "optional": false,
+            "advanced": true,
+            "allow_multiple": false,
+            "default_values": [ "biochemistry" ],
+            "field_type": "dropdown",
+            "dropdown_options": {
+                "options": [
+                    {
+                        "value": "biochemistry",
+                        "display": "Known Biochemistry",
+                        "id": "biochemistry",
+                        "ui_name": "Known Biochemistry"
+                    },
+                    {
+                        "value": "model",
+                        "display": "Input Model",
+                        "id": "model",
+                        "ui_name": "Input Model"
+                    },
+                    {
+                        "value": "none",
+                        "display": "Do not prune",
+                        "id": "none",
+                        "ui_name": "Do not prune"
+                    }
+                ]
+            }
+        },
+```
+
+There are many additional interface options available. One of the best ways to learn is to look at [this gallery](https://narrative.kbase.us/narrative/ws.23109.obj.1) which contains the a variety of KBase apps along with the spec file that generated the user interface.
+
+#### B . Configure passing variables from Narrative Input to SDK method.
+
+In the 'behavior' section of the spec.json, the output of the user interface is mapped to input to your function.
+If you have maintained a consistent naming though these mappings are pretty pro forma but you should make sure
+that you accept and return "workspace" as a narrative_system_variable as shown below.
+```
+	"behavior": {
+		"service-mapping": {
+			"url": "",
+			"name": "MegaHit",
+			"method": "run_megahit",
+			"input_mapping": [
```

Comment:
This is not the recommended way to pass object location info to an app - pass one or more references instead. @briehl can show you how to set up the method spec so that references are passed rather than names.